src/viking.js

```diff
@@ -1,11 +1,91 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor (health ,strength){
+        this.health= health,
+        this.strength= strength
+    }
+
+    attack(){
+        return this.strength;
+    }
+    receiveDamage(damage){
+        this.health = this.health - damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength); 
+        this.name = name
+        }
+    receiveDamage(damage){
+        this.health = this.health - damage;
+        return this.health > 0 ? `${this.name} has received ${damage} points of damage` : `${this.name} has died in act of combat`;
+    } 
+    battleCry(){
+        return 'Odin Owns You All!';
+    }
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage){
+        this.health = this.health - damage;
+        return this.health > 0 ? `A Saxon has received ${damage} points of damage` : `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+    this.vikingArmy = [],
+    this.saxonArmy = []
+    }
+
+    addViking(viking){
+        this.viking = { health: 300, strength: 150, name: 'Harold'};
+        this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon){
+        this.saxonArmy.push(saxon);
+    }
+    vikingAttack(){
+        let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+        let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+
+        let attackedSaxon = this.saxonArmy[randomSaxon].receiveDamage(
+          this.vikingArmy[randomViking].strength
+        );
+        if (this.saxonArmy[randomSaxon].health <= 0 ){
+         this.saxonArmy.splice(randomSaxon, 1)
+        } 
+        console.log(this.saxonArmy);
```

Comment:
When submitting finished work, make it a habit to remove any ```console.log()``` in your work. 

**Why?**
For 2 reasons:
1️⃣ The ```console.log``` doesn't add any value to the code at this point since you already have an idea of what ```this.saxonArmy``` is.
2️⃣ The ```console.log``` won't add any value if this piece of code happens to go to a production environment. When you later work in a company, there will most likely already be tests that are looking at what this value is. In that regard, you wouldn't need to have the console.log

Reply:
thanks for your feedback , will work on that and keep it in mind always